lib/node.js

```diff
@@ -1324,23 +1324,38 @@ declare class http$ServerResponse extends stream$Writable {
   writeHead(status: number, headers?: {[key: string] : string}): void;
 }
 
-declare module "http" {
-  declare class Server extends net$Server {
-    listen(port?: number, hostname?: string, backlog?: number, callback?: Function): Server;
-    // The following signatures are added to allow omitting intermediate arguments
-    listen(port?: number, backlog?: number, callback?: Function): Server;
-    listen(port?: number, hostname?: string, callback?: Function): Server;
-    listen(port?: number, callback?: Function): Server;
-    listen(path: string, callback?: Function): Server;
-    listen(handle: Object, callback?: Function): Server;
-    listening: boolean;
-    close(callback?: (error: ?Error) => mixed): Server;
-    maxHeadersCount: number;
-    keepAliveTimeout: number;
-    setTimeout(msecs: number, callback: Function): Server;
-    timeout: number;
-  }
+declare class http$Server extends net$Server {
+  listen(port?: number, hostname?: string, backlog?: number, callback?: Function): Server;
```

Comment:
Looks like all the mentions of `Server` here need to be updated. You could change them to `http$Server`, but that would cause errors if somebody did something equivalent to `(server.listen(...): Server)`. Probably best to just update all of these to `this`, which should avoid causing new errors.

Once you have this internally-consistent (i.e. running `flow` on an empty project reports no errors), let me know and I'll take a closer look.